src/main/kotlin/uk/co/craigbass/pratura/usecase/ViewBasket.kt

```diff
@@ -3,46 +3,60 @@ package uk.co.craigbass.pratura.usecase
 import uk.co.craigbass.pratura.boundary.ViewBasket
 import uk.co.craigbass.pratura.boundary.ViewBasket.*
 import uk.co.craigbass.pratura.domain.*
+import uk.co.craigbass.pratura.math.*
 import java.math.BigDecimal
 import java.math.BigDecimal.ZERO
 
 class ViewBasket(private val basketItemsGateway: BasketItemsRetriever,
                  private val productRetriever: ProductRetriever) : ViewBasket {
-  private var items: List<BasketItem>? = null
-  private var products: List<Product>? = null
-
   override fun execute(request: Unit): PresentableBasket {
-    items = basketItemsGateway.all()
-    products = productRetriever.all()
-
-    return PresentableBasket(lineItems(), "£${getBasketValue()}")
+    val pricedLineItems = pricedLineItems()
+    return PresentableBasket(
+      lineItems = getLineItems(pricedLineItems),
+      basketValue = getBasketTotal(pricedLineItems)
+    )
   }
 
-  private fun lineItems(): List<PresentableLineItem> {
-    return items!!.map { item ->
+  private fun getLineItems(pricedLineItems: List<PricedLineItem>): List<PresentableLineItem> {
```

Comment:
Generally speaking make functions that operate on single elements and do the map outside